atom/browser/api/atom_api_protocol.cc

```diff
@@ -202,42 +202,89 @@ Protocol::JsProtocolHandler Protocol::GetProtocolHandler(
   return protocol_handlers_[scheme];
 }
 
+void Protocol::OnRegisterProtocol(const std::string& scheme,
+                                  const JsProtocolHandler& handler,
+                                  const JsCompletionCallback& callback,
+                                  int is_handled) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  v8::Locker locker(isolate());
+  v8::HandleScope handle_scope(isolate());
+
+  if (is_handled || ContainsKey(protocol_handlers_, scheme)) {
```

Comment:
Since we are going to remove all locks, we should read/write `protocol_handlers_` in IO thread, in case of race conditions.